src/zope/app/apidoc/utilities.py

```diff
@@ -41,33 +42,43 @@
 _remove_html_overhead = re.compile(
     r'(?sm)^<html.*<body.*?>\n(.*)</body>\n</html>\n')
 
-space_re = re.compile('\n^( *)\S', re.M)
+space_re = re.compile(r'\n^( *)\S', re.M)
 
 _marker = object()
 
-BASEDIR = dirname(dirname(dirname(dirname(zope.app.__file__))))
 
 def relativizePath(path):
-    return path.replace(BASEDIR, 'Zope3')
-
+    matching_paths = [p for p in sys.path if path.startswith(p)]
+    if not matching_paths: # pragma: no cover
+        return path
+    longest_matching_path = max(matching_paths, key=len)
+    common_prefix = os.path.commonprefix([longest_matching_path, path])
+    return path.replace(common_prefix, 'Zope3') if common_prefix else path
```

Comment:
I can contrive examples where this would do the wrong thing, but I don't find them realistic so meh, lgtm.